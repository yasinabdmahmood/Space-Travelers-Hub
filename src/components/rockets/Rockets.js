import { useSelector, useDispatch } from 'react-redux';
import { toggleReservation } from '../../redux/rockets/rocketReducer';
import style from './rockets.module.css';

function Rockets() {
  const rockets = useSelector((state) => state.rockets);
  const dispatch = useDispatch();
  return (
    <div className={style.container} data-testid="list-of-rockets">

      {rockets.map((el) => (
        <div key={el.id} className={style.item}>
          <img src={el.flickr_images} alt="Rocket" />
          <div className={style.info}>
            <h1>{el.rocket_name}</h1>
            <p>
              <button
                type="submit"
                className={el.reserved ? style.reserved : style.hide}
              >
                Reserved
              </button>
              {el.description}
            </p>
            <button
              type="submit"
              onClick={() => dispatch(toggleReservation(el.id))}
              className={el.reserved ? style.cancel : style.myButton}
            >
              {el.reserved ? 'Cancel Reservation' : 'Reserve Rocket' }
            </button>
          </div>
        </div>
      ))}

    </div>
  );
}

export default Rockets;
