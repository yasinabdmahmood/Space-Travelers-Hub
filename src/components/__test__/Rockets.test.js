import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Rockets from '../rockets/Rockets';
import store from '../../redux/configureStore';
import '@testing-library/jest-dom';

it('Check if the component has changed', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Rockets />
    </Provider>,

  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('the component should render a div container', () => {
  render(
    <Provider store={store}>
      <Rockets />
    </Provider>,
  );

  const rocket = screen.getByTestId('list-of-rockets');

  expect(rocket).toBeInTheDocument();
});
