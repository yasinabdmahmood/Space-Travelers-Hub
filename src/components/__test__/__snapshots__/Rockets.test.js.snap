// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if the component has changed 1`] = `
<div
  className="container"
  data-testid="list-of-rockets"
/>
`;
